test/integration/node-specific/client_close.test.ts

```diff
@@ -128,25 +329,138 @@ describe.skip('MongoClient.close() Integration', () => {
+
+                await client.close();
+                expect(getMinPoolSizeTimer(servers)).to.not.exist;
+                expect(getTimerCount()).to.equal(0);
+              };
+              await runScriptAndGetProcessInfo('timer-min-pool-size', config, run);
+            });
           });
         });
 
         describe('Node.js resource: checkOut Timer', () => {
-          // waitQueueTimeoutMS
           describe('after new connection pool is created', () => {
-            it.skip('the wait queue timer is cleaned up by client.close()', async () => {});
+            it('the wait queue timer is cleaned up by client.close()', async function () {
+              const run = async function ({
+                MongoClient,
+                uri,
+                expect,
+                sinon,
+                sleep,
+                getTimerCount,
+                timers
+              }) {
+                const waitQueueTimeoutMS = 1515;
+
+                // configure failPoint
+                const utilClient = new MongoClient(uri);
+                await utilClient.connect();
+                const failPoint = {
+                  configureFailPoint: 'failCommand',
+                  mode: { times: 1 },
+                  data: {
+                    blockConnection: true,
+                    blockTimeMS: waitQueueTimeoutMS * 3,
+                    failCommands: ['insert']
+                  }
+                };
+                await utilClient.db('admin').command(failPoint);
+
+                const timeoutStartedSpy = sinon.spy(timers, 'setTimeout');
+
+                const client = new MongoClient(uri, {
+                  minPoolSize: 1,
+                  maxPoolSize: 1,
+                  waitQueueTimeoutMS
+                });
+                const insertPromise = client
+                  .db('db')
+                  .collection('collection')
+                  .insertOne({ x: 1 })
+                  .catch(e => e);
+                client
+                  .db('db')
+                  .collection('collection')
+                  .insertOne({ x: 1 })
+                  .catch(e => e);
+
+                // don't allow entire checkout timer to elapse to ensure close is called mid-timeout
+                await sleep(waitQueueTimeoutMS / 2);
+                const checkoutTimeoutStarted =
+                  timeoutStartedSpy
+                    .getCalls()
+                    .filter(r => r.args.includes(waitQueueTimeoutMS))
+                    .flat().length > 0;
+                expect(checkoutTimeoutStarted).to.be.true;
+
+                await client.close();
+                expect(getTimerCount()).to.equal(0);
+                // un-configure fail{oint
+                await utilClient.db().admin().command({
+                  configureFailPoint: 'failCommand',
+                  mode: 'off'
+                });
+
+                await utilClient.close();
+
+                const err = await insertPromise;
```

Comment:
Shouldn't the first insertPromise succeed (just after a long time), its the second one that won't be able to complete connection checkout

Reply:
I was confused / surprised about this as well. If we await both of them, the first one fails and the second one succeeds.

Reply:
Desk conversation: fixed the event awaiting